python/graphstorm/gconstruct/transform.py

```diff
@@ -665,6 +666,128 @@ def call(self, feats):
+        if self._summation is not None:
+            return {self.feat_name: None}
+
+        if isinstance(feats, ExtMemArrayWrapper):
+            # TODO(xiangsx): This is not memory efficient.
+            # It will load all data into main memory.
+            feats = feats.to_numpy()
+
+        if feats.dtype not in [np.float64, np.float32, np.float16, np.int64, \
+                              np.int32, np.int16, np.int8]:
+            logging.warning("The feature %s has to be floating points or integers,"
+                            "but get %s. Try to cast it into float32",
+                            self.feat_name, feats.dtype)
+            try:
+                # if input dtype is not float or integer, we need to cast the data
+                # into float32
+                feats = feats.astype(np.float32)
+            except: # pylint: disable=bare-except
+                raise ValueError(f"The feature {self.feat_name} has to be integers or floats.")
+
+        # make summation a 1D array
+        summation = np.sum(feats, axis=0, keepdims=True).reshape((-1,))
+        return {self.feat_name: summation}
+
+    def update_info(self, info):
+        # User has provided the sum value.
+        if self._summation is not None:
+            return self._summation
+
+        # We have to aggregative sum value from workers.
+        summations = []
+        for sum_i in info:
+            summations.append(sum_i)
+        summations = np.stack(summations)
+        summation = np.sum(summations, axis=0)
+
+        assert not np.any(summation == 0), \
+            f"The summation of values in each data column of {self.feat_name}" \
+            f"should not equal to 0, but we got {summation}"
+        self._summation = summation
+
+        # We need to save the max_val and min_val in the config object.
+        if self._conf is not None:
+            self._conf['sum'] = self._summation.tolist()
+
+    def call(self, feats) -> dict[str, np.array]:
+        """ Do normalization for feats
+
+        Parameters
+        ----------
+        feats : np array
+            Data to be normalized
+
+        Returns
+        -------
+        np.array
+        """
+        assert isinstance(feats, (np.ndarray, ExtMemArrayWrapper)), \
+            f"Feature {self._feat_name} of NumericalMinMaxTransform " \
+            "must be numpy array or ExtMemArray"
+
+        if isinstance(feats, ExtMemArrayWrapper):
+            # TODO(xiangsx): This is not memory efficient.
+            # It will load all data into main memory.
+            feats = feats.to_numpy()
+
+        if feats.dtype not in [np.float64, np.float32, np.float16, np.int64, \
+                              np.int32, np.int16, np.int8]:
+            try:
+                # if input dtype is not float or integer, we need to cast the data
+                # into float32
+                feats = feats.astype(np.float32)
+            except: # pylint: disable=bare-except
```

Comment:
I think ``except TypeError:`` will fix the pylint error here without disabling.

Reply:
Thanks